### PR TITLE
Simplify core release inputs

### LINKED ISSUE #9431: Simplify release inputs
### Housekeeping

- [X] I am a maintainer of dbt-core

### Short description

There are too many inputs when we release and we don't touch most of them.  Only require the ones that are actually needed.  The most annoying one if the sha.  We have to enter the sha we want to release but we end up merging everything into the HEAD of the branch later in the process so we're not really just releasing that sha.

### Acceptance criteria

The only inputs required are
- target branch
- version

Determine the shad of teh HEAD of the target_branch and pass that as the sha to release.

The nightly release tests must also be updated to reflect these changes.

### Suggested Tests

This is a workflow and must be tested manually.

### Impact to Other Teams

None.

### Will backports be required?

No.

### Context

_No response_

### PR DESCRIPTION
**Pull request by emmyoop**
resolves #9431 
relates to https://github.com/dbt-labs/dbt-release/issues/69

### Problem

Release inputs are long and not all of them are really necessary or add value to being flexible.

Screenshot 2024-01-23 at 12 45 52 PM

### Solution

Simplify inputs to only what's needed
Screenshot 2024-01-23 at 12 45 36 PM

Nightly tests are also updated to reflect these changes.

Test run: https://github.com/dbt-labs/dbt-core/actions/runs/7630615252
### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [ ] I have run this code in development and it appears to resolve the stated issue  
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX
- [ ] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

### CODE REVIEW

**Comment by emmyoop** [comment]
Codecov is failing but I don't know why since all I changed were workflows.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -114,32 +66,45 @@ defaults:
...
           echo Nightly release:                    ${{ inputs.nightly_release }}
 
+      - name: "Checkout target branch"
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.target_branch }}
+
+      # release-prep.yml really shouldn't take in the sha but since core + all adapters
+      # depend on it now this workaround lets us not input it manually with risk of error.
+      # The changes always get merged into the head so we can't use a specific commit fo
```
Comment (martynydbt) [inline .github/workflows/release.yml:90]: typo: for

**Review by martynydbt — APPROVED** [review]
Sounds like that retro this am was productive! Looking forward to hearing what else came of it when I'm back in.

File: .github/workflows/release.yml (hunk 1 above)
Comment (emmyoop) [inline .github/workflows/release.yml:90]:
```suggestion
      # The changes always get merged into the head so we can't use a specific commit for
```

**Pull request merged**